### DISCUSSION TITLE
[Design Decision Question] why change to recoil from valtio?

### DISCUSSION THREAD

**Discussion by jeremy-coleman** [LWJGL/lwjgl3-www#96] [General]
I've been considering moving over a huge project from mobx to valtio , so I'm really curious what prompted the change. Any insight would be appreciated! I have been skeptical of valtio's performance for large store <-> component graphs since it doesn't highjack render directly

**Comment by apostolos** [comment]
Nothing in particular, just testing out the latest state management libs. I was curious how one could model external state sync (e.g. a matchMedia query) with Recoil.

I do have some concerns about Valtio's compatibility with Concurrent Mode (which this project uses). I will let some time pass before I re-evaluate it. See if it gets adoptions from other projects, etc.

Valtio was dead-simple to use and with almost no ceremony. You will appreciate its simplicity coming from MobX and other libs. It's very small too. However I never modeled anything complicated with it, so I don't know how it holds up in larger projects.

**Comment by apostolos** [comment]
Another issue with Valtio is that it doesn't play well with React Fast Refresh (hot reload).

This is a common issue with libs that keep state "outside" React.

You can read more information about this: https://github.com/pmndrs/valtio/issues/36

**Comment by apostolos** [comment]
I'm trying to stay on the left side of this chart:
![state](https://user-images.githubusercontent.com/41679/103188417-0b4ea400-48d1-11eb-8b23-902d29ca0579.png)

**Reply by jeremy-coleman** [reply]
Cool, thank you for sharing your thoughts. I don't know why, but I never got any notifications from your replies, I just happened to randomly see it on my own profile's page feed just now. I think I try to stay on the right side of the chart. Btw, I found this lib to be really nice https://github.com/BeTomorrow/micro-observables
